library/bignum_core.c

```diff
@@ -856,16 +856,16 @@ mbedtls_mpi_uint mbedtls_mpi_core_sub_int(mbedtls_mpi_uint *X,
     return c;
 }
 
-mbedtls_mpi_uint mbedtls_mpi_core_check_zero_ct(const mbedtls_mpi_uint *A,
-                                                size_t limbs)
+mbedtls_ct_condition_t mbedtls_mpi_core_check_zero_ct(const mbedtls_mpi_uint *A,
+                                                      size_t limbs)
 {
     mbedtls_mpi_uint bits = 0;
 
     for (size_t i = 0; i < limbs; i++) {
         bits |= A[i];
```

Comment:
Actually I think that applies to a lot of `bignum_core.c`, unfortunately. But the rest is out of scope for this PR.

Reply:
As per the other PR, we need the pointer, `A`, to be volatile, to ensure the reads happen. `bits` does not need to be volatile.

Reply:
Non-blocker: I'd envisaged the function taking a "normal" pointer to non-volatile and then explicitly casting it within the function - use of volatile is more an internal implementation detail (ensuring const-time) than part of the contract (the caller doesn't care about volatile as such)? But I don't see any warnings about implicitly casting to volatile so I think it's OK.

Reply:
+1: it is unusual to see a function declared taking a `volatile` pointer when the `volatile` relates to behaviour the function wants internally

Reply:
Since it's correct, in the interests of time, I'm going to approve and leave it to Janos to update that if he wants to as next time he's in the areas as part of ongoing bignum work.